### DISCUSSION TITLE
Project structure

### DISCUSSION THREAD

**Discussion by andreweades** [pointfreeco/isowords#83] [Q&A] [answered]
This is a great demonstration of project organisation and hyper-modularisation. Something I'm trying to do myself. But I can never create a target inside a folder as you have done with all your preview apps. May I ask how you did it?

**Comment by mbrandonw** [comment]
Hey @andreweades, I'm actually not aware of anything special we are doing in this regard 🤔. I believe we were able to simply re-arrange the files/directories as we wanted from within Xcode. I believe the only thing that really matters is the target membership, which can be updated in the inspectors side bar:

<img width="265" alt="image" src="https://user-images.githubusercontent.com/135203/113916891-624bfb00-9795-11eb-95b5-2828b5bbab02.png">

What kind of problems do you run into?

**Comment by mbrandonw** [comment, marked as answer]
So I just went through this process from scratch cause I'm making a little preview app for our leaderboards and it was pretty straightforward. All I did was add a new target in Xcode, and by default that put a `LeaderboardsPreview` directory at the root. Then, in Xcode, I just dragged and dropped it to the `Previews` directory, and only had to fix two things: 1) the preview content file path was wrong, but I don't need that so deleted it from the Xcode project, and 2) the info plist path was wrong, so had to fix it by adding `Previews/` to the beginning of it. After that everything worked like normal.

Hope it helps.

**Reply by andreweades** [reply]
Thanks. It's good to know that there is no magic involved and you can easily fix up the settings.
